src/flag_gems/ops/fft_1d.py

```diff
@@ -0,0 +1,209 @@
+import torch
+import torch.profiler
+import triton
+import triton.language as tl
+
+
+@triton.jit
+def bit_reverse_kernel(real_in, imag_in, real_out, imag_out, n):
+    """先进行逆序排列: input[i] -> output[bit_reverse(i)]"""
+    tid = tl.program_id(0)
+    if tid >= n:
+        return
+
+    # 计算位数
+    temp_n = n
+    bit_count = 0
+    while temp_n > 1:
+        temp_n //= 2
+        bit_count += 1
+
+    # 计算逆序
+    idx = tid
+    rev_idx = 0
+    temp_idx = idx
+    for i in range(bit_count):
+        rev_idx = (rev_idx << 1) | (temp_idx & 1)
+        temp_idx = temp_idx >> 1
+
+    # input[idx] -> output[rev_idx]
+    val_real = tl.load(real_in + idx)
+    val_imag = tl.load(imag_in + idx)
+    tl.store(real_out + rev_idx, val_real)
+    tl.store(imag_out + rev_idx, val_imag)
+
+
+@triton.jit
+def fft_stage_kernel(real_ptr, imag_ptr, n, stage):
+    """迭代 FFT 阶段"""
+    pi = 3.141592653589793238462643383279502884  # 圆周率
+    tid = tl.program_id(0)
+
+    if tid >= n // 2:
+        return
+
+    # 计算当前阶段的参数
+    block_size = 1 << stage  # 2^stage
+    half_block = block_size // 2
+
+    # 每个线程处理一个蝶形对
+    butterfly_group = tid // half_block  # 属于第几个大块
+    pos_in_group = tid % half_block  # 在块前半部分的位置
+
+    # 计算蝶形对中两个元素的索引
+    first_idx = butterfly_group * block_size + pos_in_group
```

Comment:
obviously, `block_size` is an exponent of 2, so this `*` can be optimized to
`butterfly << stage`, right?

Reply:
Why?